app/models/log.rb

```diff
@@ -101,10 +101,15 @@ def self.pickup_count region_id
   end
 
   def self.picked_up_by(volunteer_id, complete=true, limit=nil)
-    if limit.nil?
-      Log.joins(:log_volunteers).where("log_volunteers.volunteer_id = ? AND logs.complete=? AND log_volunteers.active", volunteer_id, complete).order('"logs"."when" DESC')
+    logs = joins(:log_volunteers).
+      where("log_volunteers.volunteer_id = ? AND logs.complete=? AND log_volunteers.active", volunteer_id, complete).
+      order('"logs"."when" DESC').
+      uniq
+
+    if limit.present?
+      logs.limit(limit.to_i)
```

Comment:
This code will actually add a `LIMIT` clause to the SQL query:

```ruby
[1] pry(main)> puts Log.picked_up_by(1).to_sql
SELECT DISTINCT "logs".* FROM "logs" INNER JOIN "log_volunteers" ON "log_volunteers"."log_id" = "logs"."id" WHERE (log_volunteers.volunteer_id = 1 AND logs.complete='t' AND log_volunteers.active) ORDER BY "logs"."when" DESC
=> nil
[2] pry(main)> puts Log.picked_up_by(1, true, 10).to_sql
SELECT  DISTINCT "logs".* FROM "logs" INNER JOIN "log_volunteers" ON "log_volunteers"."log_id" = "logs"."id" WHERE (log_volunteers.volunteer_id = 1 AND logs.complete='t' AND log_volunteers.active) ORDER BY "logs"."when" DESC LIMIT 10
=> nil
```

`logs` is still an `ActiveRecord::Relation` instance when we call `limit` on it.